src/v/cluster/topic_table.cc

```diff
@@ -277,6 +277,36 @@ topic_table::apply(update_topic_properties_cmd cmd, model::offset o) {
     co_return make_error_code(errc::success);
 }
 
+ss::future<std::error_code>
+topic_table::apply(create_materialized_topic_cmd cmd, model::offset o) {
+    const model::topic_namespace& source = cmd.value;
+    const model::topic_namespace& new_materialized_topic = cmd.key;
+    if (_topics.contains(new_materialized_topic)) {
+        co_return make_error_code(errc::topic_already_exists);
+    }
+    auto tp = _topics.find(source);
+    if (tp == _topics.end()) {
+        /// Source topic must exist if attempting to create a materialized topic
+        co_return make_error_code(errc::topic_invalid_config);
```

Comment:
maybe we should add separate error code for this situation, it will be easier to debug it in future. `errc::topic_invalid_config` is very generic.